corehq/blobs/pillow.py

```diff
@@ -19,46 +14,28 @@
 
 class BlobDeletionProcessor(PillowProcessor):
 
-    def __init__(self, blob_db, bucket_base):
+    def __init__(self, blob_db):
         super(BlobDeletionProcessor, self).__init__()
         self.blob_db = blob_db
-        self.bucket_base = bucket_base
 
     def process_change(self, pillow_instance, change):
         if change.deleted:
-            bucket = join(self.bucket_base, change.id)
+            bucket = "{}/{}".format(change.meta.data_source_name, change.id)
             self.blob_db.delete(bucket=bucket)
 
 
-def get_main_blob_deletion_pillow(pillow_id):
-    """Get blob deletion pillow for the main couch database
-
-    Using the KafkaChangeFeed ties this to the main couch database.
+def get_blob_deletion_pillow(pillow_id):
+    """Get blob deletion pillow
     """
-    return _get_blob_deletion_pillow(
-        pillow_id,
-        get_db(None),
-        PillowCheckpoint('kafka-blob-deletion-pillow-checkpoint'),
-        KafkaChangeFeed(topics=[topics.META], group_id='blob-deletion-group'),
-    )
-
-
-def get_application_blob_deletion_pillow(pillow_id):
-    """Get blob deletion pillow for the apps couch database"""
-    couch_db = couch_config.get_db_for_class(Application)
-    return _get_blob_deletion_pillow(pillow_id, couch_db)
-
-
-def _get_blob_deletion_pillow(pillow_id, couch_db, checkpoint=None, change_feed=None):
-    if checkpoint is None:
-        checkpoint = PillowCheckpoint(pillow_id)
-    if change_feed is None:
-        change_feed = CouchChangeFeed(couch_db, include_docs=False)
+    checkpoint = PillowCheckpoint(pillow_id)
     return ConstructedPillow(
         name=pillow_id,
         checkpoint=checkpoint,
-        change_feed=change_feed,
-        processor=BlobDeletionProcessor(get_blob_db(), couch_db.dbname),
+        change_feed=KafkaChangeFeed(
+            topics=[topics.META, topics.APP],
```

Comment:
Are forms ever hard-deleted? If not, I think `BlobDeletionProcessor` will not delete the blobs.